projects/mci/src/mci_output.c

```diff
@@ -78,6 +81,13 @@ static void prv_handle_drive(SoftTimerId timer_id, void *context) {
     drive_command.motor_velocity =
         is_cruise ? cruise_rx_get_target_velocity() : s_velocity_lookup[drive_state];
   }
+
+  // Set current to zero if regen braking is disabled
+  // target velocity is less than actual velocity
+  if (is_regen_brake == 0 && drive_command.motor_velocity < s_velocity_lookup[drive_state]) {
```

Comment:
1) You should check `is_regen_brake == REGEN_ENABLED` (or `!is_regen_brake` if you change to `bool`), not 0
2) I don't think `s_velocity_lookup` is the way to find the actual velocity - it just tells us what target velocity we should set when pressing the throttle in the current drive state. @hewittmcg what would be the best way to find the actual current velocity? (Also setting current to 0 if target velocity < actual velocity is the correct way to disable regen braking right?)

Reply:
For current velocity, copy-pasting from my message to Faraz:
> We already periodically send the velocity to `cruise_rx.c `around line 95 of `mci_broadcast.c` using `cruise_rx_update_velocity`.  The easiest way to get the velocity in `mci_output.c` is probably to just track current velocity with a global and implement `mci_output_update_velocity` to periodically update it.  Then you can just call `mci_output_update_velocity `at the same point in `mci_broadcast.c` .

Also, yeah, `s_velocity_lookup` just maps to unobtainable values for forward/reverse and 0 for off, which lets us control the car based on current when not using cruise control.  Note that this means that when the car is going backwards, the target velocity should always be **less** than the actual velocity, which is an edge case that might need to be considered.

I read over the datasheet again and yeah, any instance where target < actual will cause regen braking if current is > 0. Because of this, I think we may want to avoid using cruise control with regen braking off, since the actual velocity could reach above the target velocity before we get a message and turn off current.  This would cause a bit of regen braking to still occur.

Reply:
I mostly implemented these.

Two points that I was unsure about were:

1- How to update actual_velocity for test cases in `test_mci_output.c`. I directly called `mci_output_actual_velocity()` and set velocity for OFF: 0.0f, DRIVE: 10.0f and REVERSE: -10.0f

2- Copying from hewitt's comment:
>I read over the datasheet again and yeah, any instance where target < actual will cause regen braking if current is > 0. Because of this, I think we may want to avoid using cruise control with regen braking off, since the actual velocity could reach above the target velocity before we get a message and turn off current. This would cause a bit of regen braking to still occur.

Does that mean that the regen braking should be enabled(no change in current) when in cruise control? I don't think the tests in `test_mci_output.c` cover the cruise control on and off mode for testing, so in case I add this feature, should the test be made in  `test_cruise_rx.c` or `test_mci_output.c` ?